ws-server/src/server.rs

```diff
@@ -313,15 +318,32 @@ async fn background_task(
 
 	// Send results back to the client.
 	tokio::spawn(async move {
+		// Received messages from the WebSocket.
+		let mut rx_item = rx.next();
+
 		while !stop_server2.shutdown_requested() {
-			if let Some(response) = rx.next().await {
-				// If websocket message send fail then terminate the connection.
-				if let Err(err) = send_ws_message(&mut sender, response).await {
-					tracing::warn!("WS send error: {}; terminate connection", err);
-					break;
+			// Ensure select is cancel-safe by fetching and storing the `rx_item` that did not finish yet.
+			// Note: Although, this is cancel-safe already, avoid using `select!` macro for future proofing.
+			match futures_util::future::select(rx_item, Delay::new(ping_interval)).await {
```

Comment:
Note: this `Delay` thing spawns a new background thread to manage timers.

Since we seem to depend on tokio here anyway, why not use the timer from tokio (which may or may not avoid the extra thread, and would def avoid the extra dependency)?

Reply:
Done! Thanks
!